pkg/stanza/operator/parser/syslog/data.go

```diff
@@ -208,6 +208,92 @@ func CreateCases(basicConfig func() *Config) ([]Case, error) {
 					"version": 1,
 				},
 				Body: `215 <86>1 2015-08-05T21:58:59.693Z 192.168.2.132 SecureAuth0 23108 ID52020 [SecureAuth@27389 UserHostAddress="192.168.2.132" Realm="SecureAuth0" UserID="Tester2" PEN="27389"] Found the user for retrieving user's profile`,
+			}},
+			true,
+			false,
+		},
+		{
+			"RFC6587 Octet Counting multi events",
+			func() *Config {
+				cfg := basicConfig()
+				cfg.Protocol = RFC5424
+				cfg.EnableOctetCounting = true
+				return cfg
+			}(),
+			&entry.Entry{
+				Body: `215 <86>1 2015-08-05T21:57:59.693Z 192.168.2.132 SecureAuth0 23108 ID52020 [SecureAuth@27389 UserHostAddress="192.168.2.132" Realm="SecureAuth0" UserID="Tester2" PEN="27389"] Found the user for retrieving user's profile` +
+					`215 <86>2 2015-08-05T21:58:59.693Z 192.168.2.133 SecureAuth0 23108 ID52020 [SecureAuth@27389 UserHostAddress="192.168.2.133" Realm="SecureAuth0" UserID="Tester2" PEN="27389"] Found the user for retrieving user's profile` +
+					`215 <86>3 2015-08-05T21:59:59.693Z 192.168.2.134 SecureAuth0 23108 ID52020 [SecureAuth@27389 UserHostAddress="192.168.2.134" Realm="SecureAuth0" UserID="Tester2" PEN="27389"] Found the user for retrieving user's profile`,
```

Comment:
I have to wonder if it would be simpler to just split the entries before feeding them into the parser 1 at a time. Seems it might just be a matter of [providing a custom split func](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7c0fd3de2aa7cbf61d6360c6208079c761992512/pkg/stanza/operator/input/tcp/tcp.go#L122C3-L122C12) for this format?

I think I would prefer this approach because input operators already have the notion of splitting a packet into multiple log entries, whereas parsers assume 1:1. I don't see a strong need to change this unless necessary. Can you help me understand if the split func approach can work, and if not, why?

Reply:
more or less, i didnt realize that is an option. ive implemented the change as a splitfunc. let me know if that works better?